update sheets-import template with new UI

shrugs opened the pull request:
sheets-import template, with Angular

Currently undergoing the 24 hour test

@erickoledadevrel

Commit:
update sheets-import template with new UI

Shrugs commented:
I wrote this code -_-

Shrugs commented:
I did write this code.

Commit:
add demo gif in README

shrugs committed:
move core libs to ajax.googleapis.com

shrugs committed:
docs: add the @AUTH block for the github-specific values

ryan-devrel commented on sheets-import/README.md:
```diff
@@ -1,102 +1,270 @@
-add-ons must go through a
-[publishing process](https://developers.google.com/apps-script/add-ons/publish)
-before they can be made available publicly.
+# sheets-import
```
The title should be consistent with the other titles in the template repo (initial upper case).  "Sheets-Import" or "Template: Importing to Sheets" or something similar would be more consistent. If you change the title, be sure it is changed everywhere in the README.

ryan-devrel commented on sheets-import/README.md:
```diff
@@ -1,102 +1,270 @@
+#### 2. Add Required Libraries to the Project
+
+  1. Click `Resources` > `Libraries`.
+  2. Paste `MswhXl8fVhTFUH_Q3UOJbXvxhMjh3Sh48` (Underscore) into the
```
You have the codes here reversed; the first one listed is the OAuth2 library, not Underscore.

ryan-devrel replied:
If it's possible, see if you can eliminate the need for the Underscore library from the template. It may make the code a little more complex, but will remove a setup step.

ryan-devrel commented on sheets-import/README.md:
```diff
@@ -1,102 +1,270 @@
+    full name (i.e. `app.js.html`).
+  - For a Script file, select `File` > `New` > `Script File` and enter the
+    full name (i.e. `server.gs`).
+  - The files you must copy over are:
```
You should add some short file descriptions here so that the developers know what purpose each file is serving.

ryan-devrel commented on sheets-import/authorize.html:
```diff
@@ -0,0 +1,76 @@
+
+angular.module('addonApp').controller('AuthCtrl', function ($scope, $rootScope, $window, $q, $goog) {
+
+  // get a new authorization url when this controller is instantiated
```
Here and everywhere else:  Comments should start capitalized and end with a period.  This will make the docs more consistent with our other documentation.

ryan-devrel commented on sheets-import/config.gs:
```diff
@@ -0,0 +1,74 @@
+ */
+
+var ADDON_NAME = 'YOUR_ADDON_NAME_HERE';
+var SIDEBAR_TITLE = 'My Super Awesome Sidebar';
```
Fun, but it's better to reference the template title in the sidebar title (such as "Import to Sheets") so that the developer is immediately aware of which add-on (or template in this case) created it. No need to include "Sidebar" in the title either, as that is immediately obvious when it is open.

ryan-devrel commented on sheets-import/index.html:
```diff
@@ -0,0 +1,102 @@
+
+  <!-- LIBS -->
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
```
If this is being included as a JS library, is it still necessary to include the Underscore Apps Script library? See if you can eliminate it.

ryan-devrel commented on sheets-import/index.html:
```diff
@@ -0,0 +1,102 @@
+      display: flex;
+      flex-direction: column;
+      min-height: 100vh;
+
```
Unnecessary blank line.

ryan-devrel commented on sheets-import/main.html:
```diff
@@ -0,0 +1,142 @@
+
+    <h3>Welcome to the Reports Dashboard</h3>
+    <p ng-if="isFirstRun" class="small">
+      With this Sheets addon, you can create automatically generated spreadsheet
```
"addon" should be hyphenated:  "add-on"

ryan-devrel commented on sheets-import/myService.gs:
```diff
@@ -0,0 +1,327 @@
+function getParams() {
+
+  /**
+   *  Params can be Text fields, Checkboxs, or Select dropdowns.
```
"Checkboxes"

ryan-devrel commented on sheets-import/myService.gs:
```diff
@@ -0,0 +1,327 @@
+}
+
+/**
+ * Return the data row, ordered correctly
```
Missing period, here and below.

ryan-devrel commented on sheets-import/myService.gs:
```diff
@@ -0,0 +1,327 @@
+ * @param {Array} row original data row, without reordering
+ * @return {Array} the final row to be entered into the spreadsheet, reordered
+ */
+function arrangDataRowForColumns(columns, row) {
```
I think you meant to title this function 'arrangeDataRowForColumns'

ryan-devrel commented on sheets-import/myService.gs:
```diff
@@ -0,0 +1,327 @@
+}
+
+/**
+ * Return the set of results after config.triggerHead
```
Missing periods again.

ryan-devrel commented on sheets-import/ngAppsScript.js.html:
```diff
@@ -0,0 +1,146 @@
+    template: '<button class="blue"><ng-transclude></ng-transclude> <i class="fa fa-check"></i></button>'
+  };
+});
+
```
No need for all these blank lines.

ryan-devrel commented on sheets-import/server.gs:
```diff
@@ -65,15 +55,8 @@ function onInstall(e) {
  * project file.
```
Make sure the project file name mentioned here is still correct.

ryan-devrel commented on sheets-import/server.gs:
```diff
@@ -81,72 +64,91 @@ function showSidebar() {
-    return 0;
-  });
-  return {reports: reportList, columns: getColumnOptions()};
+function getBootstrapData() {
```
Seems like 'getSidebarBootstrapData()' would be a little more clear.

ryan-devrel commented on sheets-import/server.gs:
```diff
@@ -81,72 +64,91 @@ function showSidebar() {
- * Return data needed to build the sidebar UI: a list of the names of the
- * currently saved report configurations and the list of potential
- * column choices.
+ * Return data needed to build the sidebar UI
```
Missing period.

ryan-devrel commented on sheets-import/server.gs:
```diff
@@ -81,72 +64,91 @@ function showSidebar() {
- * Get the report configuration for the given report and, if a sheet
- * exists for it, activate that sheet.
- * @param {String} reportId a report ID.
+ * Example of a authorization callback function that is called after an
```
Nitpick: JSDoc comments that describe functions should start with a verb.

ryan-devrel commented on sheets-import/server.gs:
```diff
@@ -81,72 +64,91 @@ function showSidebar() {
+    throw e;
   }
 
   for (var i = 1; i <= sheet.getLastColumn(); i++) {
```
Is the ss.toast() message here still needed if you have a similar toast message popping up in the sidebar?

ryan-devrel commented on sheets-import/server.gs:
```diff
@@ -157,49 +159,27 @@ function runImport(reportId) {
- * Save the given report configuration.
- * @param {Object} config a report configuration to save.
- * @return {Object} the updated report configuration.
+ * pads the 2D array to be square
```
Capitalize and add period.

ryan-devrel commented on sheets-import/server.gs:
```diff
@@ -295,25 +278,40 @@ function sendReauthorizationRequest() {
  */
 function adjustScheduleTrigger() {
+
+  if (TESTING) {
```
Since TESTING is true by default, this means triggers cannot be scheduled. I'm not sure this is the best default behavior -- what is your reasoning for including this flag? If it was simply useful while you were developing the template, consider discarding it.

ryan-devrel commented on sheets-import/util.gs:
```diff
@@ -0,0 +1,403 @@
+
+/**
+ * Includes the given template in the HTML file as content in a <template> tag
+ * with the id as the filename
```
Missing periods -- here and elsewhere.

ryan-devrel commented on sheets-import/util.gs:
```diff
@@ -0,0 +1,403 @@
+}
+
+/**
+ * Given an id and an object, save that object
```
Nitpick: reword to start with verb, here and elsewhere.

ryan-devrel commented on sheets-import/util.gs:
```diff
@@ -0,0 +1,403 @@
+}
+
+/**
+ * Given an id, eturn the (JSON-parsed) object.
```
"return" not "eturn".

ryan-devrel commented on sheets-import/util.gs:
```diff
@@ -0,0 +1,403 @@
+ * false otherwise.
+ * @param {Object} config a report configuration.
+ */
+function canEditReport(config) {
```
A function defined in a "utilities" file carries the connotation that the function is generic and can be used as is in other projects.  You've places several functions here that are specific to report creation/manipulation -- they aren't really utilities.  Consider breaking them out into a separate file.

Commit:
formatting fixes, comment improvement, and typo removal

Shrugs commented:
I don't think removing underscore is worth it; `map`, `filter`, `findWhere`, etc, are all used pretty frequently in the frontend and backend functions.

I think the `TESTING` variable should still be there (if the dev ever wants to modify the script in a non-container-bound manner), but it should be `false` by default, so I've changed it.

re: utlities, I don't think the extra file to copy/paste is worth the strict separation of concerns. I've added a block comment to separate the agnostic utility functions from the report-oriented ones.

ryan-devrel commented on sheets-import/README.md:
```diff
@@ -1,102 +1,271 @@
+    full name (i.e. `app.js.html`).
+  - For a Script file, select `File` > `New` > `Script File` and enter the
+    full name (i.e. `server.gs`).
+  - The files you must copy over (descriptions of each at the bottom of the
```
These files should have short descriptions here explaining what each file's purpose is.

ryan-devrel commented on sheets-import/README.md:
```diff
@@ -1,102 +1,271 @@
+
+The **Sheets-Import** template makes use of [AngularJS](https://angularjs.org).
+If you're not familiar with Angular, we recommend working through the
+[tutorial on the site](https://docs.angularjs.org/tutorial).
```
s/working through the [tutorial on the site]/working through this [tutorial]/

ryan-devrel commented on sheets-import/README.md:
```diff
@@ -1,102 +1,271 @@
+`report.html`, and `authorize.html`, along with the add-on global css and
+various libraries.
+
+`config.gs` contains the basic configuration options that you must modify
```
Missing periods again, here and below.

ryan-devrel commented on sheets-import/README.md:
```diff
@@ -1,102 +1,271 @@
+The **Sheets-Import** template makes use of [AngularJS](https://angularjs.org).
+If you're not familiar with Angular, we recommend working through the
+[tutorial on the site](https://docs.angularjs.org/tutorial).
+
```
I don't see ngAppsScript.js.html or app.js.html mentioned on this list; please provide a short description of those as well.

Commit:
clean up file manifest in README
